tests/MongoDB.Driver.Core.Tests/Core/Connections/ConnectionInitializerTests.cs

```diff
@@ -39,6 +39,37 @@ public ConnectionInitializerTests()
             _subject = new ConnectionInitializer("test", new [] { new CompressorConfiguration(CompressorType.Zlib) });
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void InitializeConnection_should_acquire_connectionId_from_isMaster_response([Values(false, true)] bool async)
+        {
+            var isMasterReply = MessageHelper.BuildReply(
+                RawBsonDocumentHelper.FromJson("{ ok: 1, connectionId: 1 }"));
+            var buildInfoReply = MessageHelper.BuildReply(
+                RawBsonDocumentHelper.FromJson("{ ok: 1, version: \"4.2.0\" }"));
+            var gleReply = MessageHelper.BuildReply(
```

Comment:
Is `gleReply` needed?

Reply:
@rstam , If I'm not mistaken, we can skip `gleReply` here, but in case if the PR logic will wrong and we will call `getLastError` (by any reason), we will catch unexpected error in the line https://github.com/MikalaiMazurenka/mongo-csharp-driver/blob/csharp2571/src/MongoDB.Driver.Core/Core/Connections/ConnectionInitializer.cs#L67. Then, this error will be suspended via `try/catch` and the test result won't be affected (because this assertion will still be valid: https://github.com/MikalaiMazurenka/mongo-csharp-driver/pull/17/files#diff-3a9812a69e8956908126e68c40734b31R69), but probably there is no harm to avoid this possible exception via this line too

Reply:
Fixed.